pkg/infra/teams/team1/team1_agent.go

```diff
@@ -263,6 +263,10 @@ func (s *SocialAgent) HandleUpdateShield(_ agent.BaseAgent) decision.ItemIdx {
 	return decision.ItemIdx(0)
 }
 
+func (r *SocialAgent) HandleTradeNegotiation(_ agent.BaseAgent, availableWeapons *immutable.List[state.Item], availableShields *immutable.List[state.Item], negotiations *immutable.Map[commons.TradeID, message.TradeNegotiation]) message.TradeMessage {
```

Comment:
Would be nice to reduce the signature here up by creating some struct that contains the availableLoot and negotiations similar to what I did for the internal package but in this case one that's exported and external.